### PR TITLE
Introduce an E2E test: calling OpenAI for structured-output :test_tube: :calling: 

### PR DESCRIPTION
**Pull request by ivy-rew**
As discussed; a first end2end test for OpenAI :
- by intention we fire only a single test with very deterministic results ... the goal is not to cover every corner case of the demos. But to use once the real LLM over its official endpoint, with zero mocks in charge.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  contents: read
+  checks: write
+
+jobs:
+  build:
+    uses: axonivy-market/github-workflows/.github/workflows/ci.yml@v6
+    with:
+      mvnArgs: -P e2e.test 
+    secrets:
+      OPEN_AI_API_KEY: ${{ secrets.OPEN_AI_API_KEY }}
```
Comment (ivy-rew) [inline .github/workflows/e2e.yml]: consume from github secrets

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: E2E-Build
+
+on:
+  schedule:
+    - cron:  '0 1 * * 0' # 1 AM on Sunday
+  workflow_dispatch:
```
Comment (ivy-rew) [inline .github/workflows/e2e.yml]: on request, or once a week

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  
+permissions:
+  contents: read
+  checks: write
+
+jobs:
+  build:
+    uses: axonivy-market/github-workflows/.github/workflows/ci.yml@v6
+    with:
+      mvnArgs: -P e2e.test
```
Comment (ivy-rew) [inline .github/workflows/e2e.yml]: only enable e2e.test via maven profile

File: smart-workflow-test/src_test/com/axonivy/utils/smart/workflow/model/openai/OpenAiModelIT.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  @Test
+  void structuredOutput_e2e(BpmClient client) {
+    Ivy.session().loginSessionUser("James", "secret");
+    var res = client.start()
+        .process(AGENT_TOOLS.elementName("structuredOutput"))
+        .as().session(Ivy.session())
+        .execute();
+    TestToolUserData data = res.data().last();
+    assertThat(data.getPerson().getFirstName())
+        .isEqualTo("James");
```
Comment (ivy-rew) [inline smart-workflow-test/src_test/com/axonivy/utils/smart/workflow/model/openai/OpenAiModelIT.java]: a test which has it's complexity, calling tools, but still should provide very deterministic results, as we enforce a structured output of person and produce the tool-value on our terms.

... I guess the same test can be run against every other model-provider with the same results :crossed_fingers:

**Review by mnhnam-axonivy — APPROVED** [review]
LGTM please merge 👍
